umh.docs.umh.app/content/en/docs/datacontracts/_index.md

```diff
@@ -0,0 +1,213 @@
+---
+title: "Data Contracts / API"
+menuTitle: "Data Contracts / API"
+description: "This page describes how messages flow in the UMH, which message goes where, how it has to be formatted and how you can create your own structures."
+weight: 1725
+---
+
+## What are Data Contracts
+
+Data Contracts are formal agreements that define the structure, format, and
+rules for data exchanged between components within a Unified Namespace (UNS)
+architecture. They specify metadata, data models, and service levels to ensure
+that all systems interact consistently and reliably. By establishing clear
+guidelines for data exchange, Data Contracts facilitate seamless integration
+and prevent disruptions caused by changes in data formats or structures.
+
+In short, they define where a message is going, what structure it must follow,
+how it gets there, and what happens when it arrives, based on defined rules and
+services.
+
+### Example Historian
+
+To give you a simple example, just think about the `_historian' schema. Perhaps
```

Comment:
general note about _historian:

the timestamp makes each message idempotent. every message is transmitted using "at least once", which can cause messages being duplicated somewherei n the system. by having it idempotent, it doesn't matter how often they get sent, they will only land up once in the database